### ISSUE TITLE
Accented characters through wincompose toggle play/pause on soundcloud

### ISSUE THREAD

**Issue by benoit-germain** [carlin-q-scott/browser-media-players#20]
I use [wincompose](https://github.com/samhocevar/wincompose) to generate accented characters and other useful symbols, such as à,ê, and →. Whenever I issue such a character, soundcloud is toggled on and off, as if browser-media-players detected a media key press event.
🦁, however, leaves soundcloud alone.
Typing a é using a french keyboard layout doesn't exhibit the issue.
I don't know if wincompose is to blame, I posted a similar issue to it.

**Comment by carlin-q-scott** [comment]
I wouldn't be surprised if the culprit is Firefox itself. Firefox [combines most if not all of the extended keycodes](https://bugzilla.mozilla.org/show_bug.cgi?id=1251795#c14), so anything that isn't on a normal keyboard will probably trigger the play/pause event for this extension.

Would #9 be an acceptable workaround until the issue in Firefox is addressed? I've been planning to fix it myself but their code is difficult to figure out and it takes an hour to compile each time I make a change.

**Comment by samhocevar** [comment]
WinCompose works by using `SendInput` with `KEYEVENTF_UNICODE` which, [according to the documentation](https://docs.microsoft.com/en-us/windows/desktop/api/winuser/ns-winuser-tagkeybdinput), sends a `WM_KEYDOWN` and a `WM_KEYUP` message with `VK_PACKET` as their parameter.

I could confirm that Firefox properly receives these `VK_PACKET` messages on my system. However, as soon as `browser-media-players` is activated (no need to be on a media site, I just activate it in `about:addons`) then it stops receiving it when I type *e.g.* in the URL bar.

**Comment by samhocevar** [comment]
FYI, the low-level part of Firefox (`KeyboardLayout.cpp`) seems to work; I can confirm that JavaScript `keydown` events properly decode WinCompose input, even with `browser-media-players` activated (see [this demo](https://codepen.io/anon/pen/vPzxEo)). That input vanishes when typing in the URL bar when the plugin is activated, though.

I am severely JavaScript illiterate so I can’t really help much more with debugging, but for the moment I would say `browser-media-players` is eating events that it shouldn’t be.

**Comment by carlin-q-scott** [comment]
Yeah, Firefox is providing the events to the JavaScript engine just fine, but the [Commands key binding system provided to WebExtensions](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/commands) isn't connecting them correctly. Those are specified in the manifest.json for browser-media-players and are the root of the issue.

You can do a similar test to the codepen test you shared by using the options page for browser-media-players where it lets you set custom key bindings.

**Comment by benoit-germain** [comment]
I'm out of my depth here, but to answer your question, yes, #9 would be just fine with me as a workaround :-).

**Comment by carlin-q-scott** [comment]
This is a slight variation of #11

**Comment by carlin-q-scott** [comment]
@benoit-germain I finally released feature #9.

**Comment by benoit-germain** [comment]
Thanks!
